### PR TITLE
docs: allow "Neovim" and "neovim"

### PR DESCRIPTION
**Pull request by dundargoc**
No one cares about the correct usage between nvim, Nvim, Neovim, or
neovim. They are all used interchangably.

### CODE REVIEW

**Comment by dundargoc** [comment]
Also makes writinf docs super annoying in cases where the usage of Neovim is correct

**Comment by zeertzjq** [comment]
> Also makes writinf docs super annoying in cases where the usage of Neovim is correct

What cases?

**Comment by dundargoc** [comment]
All the cases where it is correct.

Does listing these somehow change your mind about anything?

File: scripts/gen_help_html.lua [hunk 1]
```diff
@@ -35,9 +35,7 @@ local invalid_links = {} ---@type table<string, any>
 local invalid_urls = {} ---@type table<string, any>
 local invalid_spelling = {} ---@type table<string, table<string, string>>
 local spell_dict = {
-  Neovim = 'Nvim',
   NeoVim = 'Nvim',
-  neovim = 'Nvim',
```
Comment (gpanders) [inline scripts/gen_help_html.lua:40]: What is the case for allowing `neovim`? When is it ever correct to use `neovim` (as opposed to "Neovim", which at least has correct capitalization)?

Comment (dundargoc) [reply scripts/gen_help_html.lua:40]: I don't see why both can't be correct

**Comment by wookayin** [comment]
> `master`/runtime%2Fdoc%2Fbackers.txt#L10

https://github.com/neovim/neovim/blob/v0.10.0/scripts/gen_help_html.lua#L51

We have this. Wouldn't it suffice?

**Pull request closed without merging**